src/components/shared/Extensions/Accordion/partials/AccordionHeader.tsx

```diff
@@ -1,24 +1,41 @@
 import React, { FC } from 'react';
+import clsx from 'clsx';
+import { useIntl } from 'react-intl';
 
 import { AccordionItemProps } from '../types';
 import Icon from '~shared/Icon';
 
 const displayName = 'Extensions.Accordion.partials.AccordionHeader';
 
-const AccordionHeader: FC<AccordionItemProps> = ({ title, isOpen }) => (
-  <div>
-    <div className="flex justify-between items-center pb-2 pt-4">
-      {title}
-      <Icon
-        appearance={{ size: 'extraTiny' }}
-        name={isOpen ? 'caret-up' : 'caret-down'}
-        title={{ id: 'file-text' }}
-        className="text-gray-400 min-w-[0.875rem] min-h-[0.875rem]"
-      />
-    </div>
-    <div className="w-full bg-gray-200 h-[1px]" />
-  </div>
-);
+const AccordionHeader: FC<AccordionItemProps> = ({
+  title = '',
+  isOpen,
+  mode = 'primary',
+}) => {
+  const { formatMessage } = useIntl();
+  const text = typeof title === 'string' ? formatMessage({ id: title }) : title;
+
+  return (
+    <>
+      <div
+        className={clsx('flex justify-between items-center pb-2 pt-4 w-full', {
+          'text-gray-700': mode === 'primary',
+          'text-gray-400 [&>i>svg]:text-gray-700': mode === 'secondary',
```

Comment:
We should avoid to style using nested tags, please add this color directly to the Icon container (I've added the comment below)